clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -568,17 +568,11 @@ static void collectSYCLAttributes(Sema &S, FunctionDecl *FD,
                SYCLIntelNoGlobalWorkOffsetAttr, SYCLSimdAttr>(A);
   });
 
-  // Allow the kernel attribute "use_stall_enable_clusters" only on lambda
+  // Allow the function attribute "use_stall_enable_clusters" only on lambda
```

Comment:
Is this correct? Based on this comment https://github.com/intel/llvm/pull/3900#discussion_r648460069 (and the following conversation), I thought the consensus was to leave the attribute attached to `operator()` and not propagate to kernel?

Reply:
Are you talking about the message here? 

Do not need to call DirectlyCalled here? I apologize if i misunderstood. 


```
struct FuncObj {  [[intel::use_stall_enable_clusters]] void operator()() const {}  


// Test attribute is propagated to the kernel.    // CHECK-LABEL: FunctionDecl {{.*}}test_kernel1    // CHECK:       SYCLIntelUseStallEnableClustersAttr {{.*}}    h.single_task&lt;class test_kernel1&gt;(        FuncObj());
```
is this expected behavior?

Reply:
Based on previous clarifications by other reviewers to my question regarding this (https://github.com/intel/llvm/pull/3900#discussion_r648450605)- 

>Our kernel should still be calling operator(), so we can leave it attached to operator() is my understanding. This is basically just 'not' a SYCL attribute, it is a general language attribute, it doesn't impact the kernel itself is my understanding (https://github.com/intel/llvm/pull/3900#discussion_r648460069).

and 

> I understood that to mean "make this like every other function attribute instead of like the special SYCL ones" and so there's no reason for the docs to mention kernels at all or for SemaSYCL.cpp to move the attribute onto anything else. (https://github.com/intel/llvm/pull/3900#discussion_r648469610)

it is my understanding we don't need any special behavior for this attribute and you can leave attribute attached to `operator()` without propagating it to kernel. That said, I am not sure of the requirements.

Reply:
Ok. Thanks for pointing the previous comments.